src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,284 @@
+package duke;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.format.TextStyle;
+import java.time.temporal.ChronoField;
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.exception.DukeFormatCommandException;
+import duke.exception.DukeIndexErrorException;
+import duke.exception.DukeInvalidCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Represents the main Duke class and runs MakiBot.
+ *
+ * @author Justin Peng
+ */
+public class Duke {
+    /** Datetime formatter for user input which accepts dd/MM/yyyy or dd/MM/yyyy HH:mm */
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = new DateTimeFormatterBuilder()
+            .appendPattern("dd/MM/yyyy ")
+            .optionalStart()
+            .appendPattern("HH:mm ")
+            .optionalEnd()
+            .appendZoneText(TextStyle.SHORT)
+            .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+            .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+            .parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0)
+            .toFormatter();
+    /** Commands accepted by Duke */
+    public enum Command {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, SAVE, FIND
+    }
+    /** User timezone */
+    private ZoneId timeZone = ZoneId.of("GMT+00:00");
+
+    private final Storage storage;
+    private final Ui ui;
+    private final Parser parser;
+    private TaskList tasks;
+
+    /**
+     * Creates a new {@code Duke} object.
+     */
+    protected Duke() {
+        ui = new Ui();
+        storage = new Storage();
+        parser = new Parser();
+    }
+
+    public static void main(String[] args) {
+        new Duke().start();
+    }
+
+    /**
+     * Starts MakiBot.
+     */
+    protected void start() {
+        System.out.println("Hello! I'm MakiBot");
```

Comment:
Given the bot's called MakiBot, updating the project name would really make sense!